### PR TITLE
KAZOO-3992: fix token_auth DELETE and add GET

### PR DESCRIPTION
**Pull request by k-anderson**
This will need to be backported to 3.19+

### CODE REVIEW

File: applications/crossbar/src/modules/cb_token_auth.erl [hunk 1]
```diff
@@ -51,13 +53,24 @@ init() ->
...
+allowed_methods() -> [?HTTP_DELETE, ?HTTP_GET].
 
 -spec resource_exists() -> 'true'.
 resource_exists() -> 'true'.
 
 -spec validate(cb_context:context()) -> cb_context:context().
 validate(Context) ->
+    validate(Context, cb_context:req_verb(Context)).
+
+validate(Context, <<"GET">>) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_token_auth.erl]: These should be the macros `?HTTP_GET` and `?HTTP_DELETE`

**Comment by k-anderson** [comment]
@jamesaimonetti 
PR updated!

File: applications/crossbar/src/modules/cb_token_auth.erl [hunk 2]
```diff
@@ -51,13 +53,24 @@ init() ->
...
+allowed_methods() -> [?HTTP_DELETE, ?HTTP_GET].
 
 -spec resource_exists() -> 'true'.
 resource_exists() -> 'true'.
 
 -spec validate(cb_context:context()) -> cb_context:context().
 validate(Context) ->
+    validate(Context, cb_context:req_verb(Context)).
+
+validate(Context, ?HTTP_GET) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_token_auth.erl]: Yo dawg, missing spec here

File: applications/crossbar/src/modules/cb_token_auth.erl [hunk 3]
```diff
@@ -51,13 +53,24 @@ init() ->
...
+    JObj = crossbar_util:response_auth(
+             wh_json:public_fields(cb_context:auth_doc(Context))
+            ),
+    Setters = [{fun cb_context:set_resp_status/2, 'success'}
+               ,{fun cb_context:set_resp_data/2, JObj}
+              ],
+    cb_context:setters(Context, Setters);
+validate(Context, <<"DELETE">>) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_token_auth.erl]: Use `?HTTP_DELETE` s'il te plait @k-anderson

**Comment by k-anderson** [comment]
@jamesaimonetti 
Oops, PR re-updated

**Comment by jamesaimonetti** [comment]
@k-anderson don't forget to backport

**Pull request merged**